modules/operations/continue.js

```diff
@@ -16,20 +18,27 @@ export function operationContinue(context, selectedIDs) {
 
     function candidateWays() {
         return _vertex ? context.graph().parentWays(_vertex).filter(function(parent) {
-            return parent.geometry(context.graph()) === 'line' &&
-                !parent.isClosed() &&
-                parent.affix(_vertex.id) &&
+            return parent.contains(_vertex.id) &&
                 (_geometries.line.length === 0 || _geometries.line[0] === parent);
         }) : [];
     }
 
     var _candidates = candidateWays();
+    var _candidate = _candidates[0];
+    var _candidateGeometry = _candidate && _candidate.geometry(context.graph());
 
 
     var operation = function() {
-        var candidate = _candidates[0];
+        var affix = _candidate.affix(_vertex.id);
+        // Unless an endpoint is selected, delete the selected node in favor of whatever we're going to draw.
+        // This avoids a situation where it's possible to draw along all but one of the edges.
+        if (!affix) {
+            context.perform(actionDeleteNode(_vertex.id));
```

Comment:
If you select a vertex other than an endpoint, the operation replaces it with the newly drawn vertices. Otherwise, if we persist the selected vertex and only continue drawing along either adjacent edge, then either the first or the last edge of the way becomes impossible to continue drawing along.

Ideally, this operation would detach the way from the node and only delete the node if it has no tags and no parents.